### ISSUE TITLE
pydantic BaseModel E0611

### ISSUE THREAD

**Issue by vainaixr** [PyCQA/pylint#7486]
### Bug description

I do
```
from pydantic import BaseModel
```
pylint says
```
E0611: No name 'BaseModel' in module 'pydantic' (no-name-in-module)
```
plus I do,
```
class UserModel(BaseModel):
    name: str
    age: int
```
it says R0903, too few public methods

### Command used

```shell
pylint file_name.py
```

### Pylint output

```shell
E0611: No name 'BaseModel' in module 'pydantic' (no-name-in-module)
R0903: Too few public methods (too-few-public-methods)
```

### Expected behavior

there should be no issue with importing BaseModel from pydantic
if you see the way pydantic is used, both the issues should not be there

### Pylint version

```shell
pylint 2.15.2
astroid 2.12.9
Python 3.11.0rc2
```

**Comment by mbyrnepr2** [comment]
Thank you for the report @vainaixr.

Using the following option suppresses both issues reported by your example:

```python
pylint  --extension-pkg-whitelist='pydantic' example.py
```

**Comment by vainaixr** [comment]
even SQLModel uses similar class attributes convention, do I keep adding all libraries to whitelist.
the whitelist solution is a workaround, could the first one that is no name in the module be solved?
the second one is related to creating a class with only class attributes in it, I think so, that should also be allowed, considering even more libraries use a similar convention.

**Comment by Pierre-Sassoulas** [comment]
> the whitelist solution is a workaround

You can also use ``--unsafe-load-any-extension=y``. Be aware though that using this flag means that extensions are loaded into the active Python interpreter and may run arbitrary code, which you may not want. This is the reason why we disable by default loading C extensions. 

You can disable ``too-few-public-methods`` if you don't like it, we're also going to make it optional in #3512
